packages/rulesets/src/asyncapi/functions/asyncApi2ChannelServers.ts

```diff
@@ -0,0 +1,54 @@
+import { createRulesetFunction } from '@stoplight/spectral-core';
+
+import type { IFunctionResult } from '@stoplight/spectral-core';
+
+export default createRulesetFunction<
+  { servers?: Record<string, unknown>; channels?: Record<string, { servers?: Array<string> }> },
+  null
+>(
+  {
+    input: {
+      type: 'object',
+      properties: {
+        servers: {
+          type: 'object',
+        },
+        channels: {
+          type: 'object',
+          additionalProperties: {
+            type: 'object',
+            properties: {
+              servers: {
+                type: 'array',
+                items: {
+                  type: 'string',
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+    options: null,
+  },
+  function asyncApi2ChannelServers(targetVal, _) {
+    const results: IFunctionResult[] = [];
+    if (!targetVal.channels) return results;
+    const serverNames = Object.keys(targetVal.servers ?? {});
```

Comment:
What if there are no servers declared but the channels declares a value under `servers` channel field?